### PR TITLE
Feature: Revamp Report a bug page with recaptcha and private bug reporting

### PR DESCRIPTION
**Pull request by AtmegaBuzz**
New Report a bug page #1231

image

# Preview
image

### CODE REVIEW

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
Why do we need a recpacha, the simple captcha is working fine?  Also my main issue with this new design is that it's too large for the screen - can you make everything fit on one page above the fold?

**Comment by AtmegaBuzz** [comment]
Simple captcha is bit annoying, you have to enter every time but recaptcha it just easy for users, just one tick and done if bot comes the puzzle will be asked.
Ok got it I'll fit it in one page, will move the editor to right and other things to left.

**Comment by DonnieBLT** [comment]
Can you show the full screenshot with the browser as well?  It looks good but maybe it could be a bit smaller all around

**Comment by AtmegaBuzz** [comment]
OK I'll add the suggestions, for preview I'll add a pop up when we click on the image box in red color it will show the preview and about mobile screem I will just flex wrap it, just to be sure that desktop screen looks good before moving

**Comment by AtmegaBuzz** [comment]
image
image
image
image
image
image

@DonnieBLT @just-ary27 I've added responsiveness, for more detail view you can see the video shared on slack.

File: blt/urls.py [hunk 1]
```diff
@@ -296,7 +301,8 @@
     re_path(r"^api/v1/leaderboard/$",LeaderboardApiViewSet.as_view(),name="leaderboard"),
 
     re_path(r"^scoreboard/$", ScoreboardView.as_view(), name="scoreboard"),
-    re_path(r"^issue/$", IssueCreate.as_view(), name="issue"),
+    re_path(r"^report/$", IssueCreate.as_view(), name="report"),
```
Comment (DonnieBLT) [inline blt/urls.py:304]: can we leave issue/ as there may be previous links to it - like the extension

File: website/alternative_views.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            url = "https://api.github.com/repos/%s/%s/issues" % (p.owner, p.repo)
+
+            if not data["user"]:
+                the_user = "Anonymous"
+            else:
+                the_user = data["user"]
+                
+            gh_issue = {
+                "title": issue.description,
+                "body": screenshot_text +
```
Comment (DonnieBLT) [inline website/alternative_views.py:205]: add description_text here

File: website/models.py [hunk 3]
```diff
@@ -238,8 +237,8 @@ def hostname_domain(self):
     @property
     def domain_name(self):
         parsed_url = urlparse(self.url)
-        domain = parsed_url.hostname
```
Comment (DonnieBLT) [inline website/models.py:241]: curious as to why this was changed?

File: website/static/css/navbar.css [hunk 4]
```diff
@@ -175,4 +175,4 @@
         align-items: center;
         height: 100%;
     }
-}
```
Comment (DonnieBLT) [inline website/static/css/navbar.css]: remove unchanged

File: blt/urls.py [hunk 5]
```diff
@@ -107,7 +112,7 @@
 handler500 = "website.views.handler500"
 
 urlpatterns = [
-    path("captcha/", include("captcha.urls")),
+    # path("captcha/", include("captcha.urls")),
```
Comment (DonnieBLT) [inline blt/urls.py:115]: please keep the existing captcha code so the existing report an issue page can continue to work

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
a few changes requested

File: blt/urls.py (hunk 5 above)
Comment (DonnieBLT) [reply blt/urls.py:115]: @AtmegaBuzz please revert this line

File: blt/urls.py [hunk 6]
```diff
@@ -351,7 +357,6 @@
     ),
     re_path(r"^social/$", TemplateView.as_view(template_name="social.html")),
     re_path(r"^search/$", website.views.search),
-    re_path(r"^report/$", IssueCreate.as_view()),
```
[the remark this answers is not in the document]
Comment (DonnieBLT) [reply blt/urls.py:354]: @AtmegaBuzz please add this back

File: website/models.py [hunk 7]
```diff
@@ -203,7 +202,7 @@ class Issue(models.Model):
     domain = models.ForeignKey(Domain, null=True, blank=True, on_delete=models.CASCADE)
     url = models.URLField()
     description = models.TextField()
-    captcha = CaptchaField()
```
[the remark this answers is not in the document]
Comment (DonnieBLT) [reply website/models.py:206]: @AtmegaBuzz please add this back

File: website/models.py (hunk 3 above)
Comment (DonnieBLT) [reply website/models.py:241]: @AtmegaBuzz can you comment on why this changed?

**Review by DonnieBLT — CHANGES_REQUESTED** [review]

File: website/models.py (hunk 3 above)
Comment (DonnieBLT) [reply website/models.py:241]: @AtmegaBuzz this is meant to extract the domain name from the url that was reported, it looks like it was operating correctly.  Can you please revert this?  If there is a specific case where it has to change, can you please add an automated test to show that it's working as intended?

File: poetry.lock
Comment (DonnieBLT) [inline poetry.lock:1]: Can undo all changes to the poetry files

File: blt/settings.py [hunk 8]
```diff
@@ -84,7 +84,7 @@
     # "tellme",
     "star_ratings",
     "drf_yasg",
-    "captcha",
+    'captcha',
```
Comment (DonnieBLT) [inline blt/settings.py:87]: can revert this to double quotes

File: pyproject.toml
Comment (DonnieBLT) [inline pyproject.toml:1]: can undo this change

File: website/models.py [hunk 9]
```diff
@@ -18,8 +18,8 @@
 from django.dispatch import receiver
 from rest_framework.authtoken.models import Token
 from mdeditor.fields import MDTextField
-from decimal import Decimal
```
Comment (DonnieBLT) [inline website/models.py:21]: swap this back

File: website/static/css/navbar.css [hunk 10]
```diff
@@ -175,4 +175,4 @@
...
+}
```
Comment (DonnieBLT) [inline website/static/css/navbar.css:178]: remove blank change

File: website/templates/500.html [hunk 11]
```diff
@@ -15,7 +15,7 @@ <h1>SERVER ERROR</h1>
         $(document).ready(function () {
             $(document).on('click', '.errors', function () {
                 var url = window.location.href;
-                var redirect_url = 'https://{% env "FQDN" %}/report/?url=' + url;
```
Comment (DonnieBLT) [inline website/templates/500.html:18]: /issue/ is used for the issue - can revert this

File: website/templates/includes/footer.html [hunk 12]
```diff
@@ -2,7 +2,7 @@
 {% load i18n %}
 {% load custom_tags %}
 {% comment %} 'img/logo_white.png' {% endcomment %}
-<footer class="bg-black text-white pt-12 pb-8 px-12 bottom-0 relative fixed t-0">
+<footer class="bg-black text-white pt-12 pb-8 px-12 relative ">
```
Comment (DonnieBLT) [inline website/templates/includes/footer.html:5]: what does this do? is it related to the add a new issue page?

File: website/templates/includes/navbar.html [hunk 13]
```diff
@@ -18,7 +18,7 @@
                 <!-- Dropdown for Settings Icon -->
                 <ul class="dropdown-menu left-1">
                     <li>
-                        <a href="/report">
```
Comment (DonnieBLT) [inline website/templates/includes/navbar.html:21]: revert this

File: website/templates/includes/navbar.html [hunk 14]
```diff
@@ -65,7 +65,7 @@
             <div class="items-center w-full h-full nav-hidden lg:!block">
                 <i class="fa fa-search relative left-[28px] top-[-2px] text-gray-700 text-base nav-hidden lg:!inline-block"></i>
                 <input type="text" name="query" placeholder="{% trans "Search" %}" class="w-[200px] h-[38px] pl-[30px] nav-hidden lg:!inline-block">
-                <a href="/report/" class="nav-hidden lg:!inline-block bg-red-500 hover:bg-red-600 h-[38px] py-[8px] px-[25px] whitespace-nowrap text-2xl font-bold text-white leading-[21px] cursor-pointer hover:text-white hover:no-underline">
```
Comment (DonnieBLT) [inline website/templates/includes/navbar.html:68]: revert

File: website/templates/report.html [hunk 15]
```diff
@@ -138,13 +138,12 @@ <h1 class="title">REPORT A BUG</h1>
                 <br>
                 <span>
                     <input type="file" required class="required" id='screenshot' multiple name="screenshots" />
-                    <!-- <button class="btn btn-primary btn-block"
+                    {% comment %} <button class="btn btn-primary btn-block"
```
Comment (DonnieBLT) [inline website/templates/report.html:141]: save this for a different PR

File: website/tests.py [hunk 16]
```diff
@@ -67,7 +67,7 @@ def test_post_bug(self):
         WebDriverWait(self.selenium, 30).until(
             EC.presence_of_element_located((By.TAG_NAME, "body"))
         )
-        self.selenium.get('%s%s' % (self.live_server_url, '/report/'))
+        self.selenium.get('%s%s' % (self.live_server_url, '/issue/'))
```
Comment (DonnieBLT) [inline website/tests.py:70]: revert

File: website/views.py [hunk 17]
```diff
@@ -559,7 +561,7 @@ def form_valid(self, form):
             messages.error(self.request, "Invalid Captcha!")
             return HttpResponseRedirect("/issue/")
         
-        clean_domain = obj.domain_name.replace("www.", "").replace("https://","").replace("http://","")
+        clean_domain = self.request.POST["url"].replace("www.", "").replace("https://","").replace("http://","")
```
Comment (DonnieBLT) [inline website/views.py:564]: revert

File: website/views.py [hunk 18]
```diff
@@ -624,7 +626,7 @@ def form_valid(self, form):
 
             user_prof.save()
 
-        redirect_url = "/report"
+        redirect_url = "/issue"
```
Comment (DonnieBLT) [inline website/views.py:629]: revert

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
a bit more detailed feedback

File: website/models.py (hunk 3 above)
Comment (DonnieBLT) [reply website/models.py:241]: @AtmegaBuzz please revert this unless you add a test to make sure it works

File: website/static/css/navbar.css (hunk 10 above)
Comment (DonnieBLT) [inline website/static/css/navbar.css:178]: still seeing a "no change" change here...

File: website/templates/includes/footer.html (hunk 12 above)
Comment (DonnieBLT) [reply website/templates/includes/footer.html:5]: @AtmegaBuzz what does this do?  Is it needed for the add a new issue page?  Please revert it if not - would like a minimal amount of changes files

File: website/templates/report.html [hunk 19]
```diff
@@ -138,13 +138,12 @@ <h1 class="title">REPORT A BUG</h1>
                 <br>
                 <span>
                     <input type="file" required class="required" id='screenshot' multiple name="screenshots" />
-                    <!-- <button class="btn btn-primary btn-block"
+                    <button class="btn btn-primary btn-block"
```
Comment (DonnieBLT) [inline website/templates/report.html:141]: what does this change do, can you revert it?

**Review by DonnieBLT — CHANGES_REQUESTED** [review]

**Pull request closed without merging**